src/js/jsx/search/SearchBar.jsx

```diff
@@ -194,6 +194,12 @@ define(function (require, exports, module) {
             var updatedFilter = id ? _.uniq(this.state.filter.concat(filterValues)) : [],
                 filterIcon = id && this.getFlux().store("search").getSVGClass(updatedFilter);
             
+            var nextHasInputValue = id !== null;
+            
+            if (nextHasInputValue !== this.state.hasInputValue) {
+                this.setState({ hasInputValue: nextHasInputValue });
```

Comment:
Instead of calling setState here, can you add this key-value pair to the setState call that's made below?